src/authkit-callback-route.ts

```diff
@@ -29,7 +29,19 @@ export function handleAuth(options: HandleAuthOptions = {}) {
         url.searchParams.delete('state');
 
         // Redirect to the requested path and store the session
-        url.pathname = returnPathname ?? returnPathnameOption;
+        returnPathname = returnPathname ?? returnPathnameOption;
+
+        // Extract the search params if they are present
+        if (returnPathname.includes('?')) {
```

Comment:
instead of manually splitting the params out of the url, can we access all the existing query params using searchParams too? Something like:

```!js
for (const [key, value] of new URL(returnPathname, "https://example.com").searchParams) {
  url.searchParams.append(key, value)
}
```